api/schema/aws_account_resource.go

```diff
@@ -66,6 +66,33 @@ var (
 						Mode: ImmutableAttributeMode,
 					},
 				},
+				"role_arn": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						Description: "Provider AWS role ARN.",
+						Required:    true,
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: WriteOnlyOnceAttributeMode,
+					},
+				},
+				"management_account_id": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						Description: "Management account ID.",
+						Optional:    true,
+					},
+					attributeWithMode: attributeWithMode{
+						Mode: WriteOnlyOnceAttributeMode,
+					},
+				},
+				"organization_id": StringResourceAttributeWithMode{
+					StringAttribute: resource_schema.StringAttribute{
+						Description: "AWS organization ID.",
+						Optional:    true,
```

Comment:
Add a validator to the schema to verify that either both `management_account_id` and `organization_id` are specified, or neither is.

Reply:
sure. Also i have another senario, we need to support org onboarding.
which will need to take in one master_account_id, org_id and multiple children account_ids.

which means creating multiple iam roles. and calling callback to the same aws integration record.

## *approach1:*

resource "illumio-cloudsecure_aws_account" "example" {
  account_ids           = ["123456789012", "123456789013"]
  account_type          = "Organization"
  name                  = "My AWS Account"
  service_account_id    = "service-account-id"
  organization_id       = "organization-id"
  management_account_id = "management-account-id"

  \# Optional attributes
  mode = "ReadWrite"
}


## *approach2:*
or we can facilitate it in the module. but given  and the root account id and org ids are present in the block.
so we check if it already exists if yes we just do call back.


NOTE:

cloudformation what it does not:
1. creates role,
2. creates lambda and triggers our "callback url" [its a different domain] to update the rolearn, and externalid(svcid)
3. for orgs same thing happens on the root account and for child accounts stackset is created which runs on the rest of the accounts. (one cool feature is it keeps monitoring for the new accounts getting created.) if we can somehow get it on tf too it will be  great. but alternative is run the tf script again and it will do it on the new account.

Reply:
Adding this validation will be tricky. The resources generated by `providergen` will have to implement the `ResourceWithConfigValidators` interface in addition to `ResourceWithConfigure` and `ResourceWithImportState`.
Doing this in this PR would delay it for too long.
I can take care of adding support for resource validators and this particular validation later.